### PR TITLE
developer doc: how to build a filter

### PR DESCRIPTION
**Pull request by szuecs**
#751 Adds how to build a filter with docs, tests and implementation.

Signed-off-by: Sandor Szuecs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+If a filter, predicate or dataclient need `Options` passed from flags,
+then you should register the filter in `skipper.go`, the main library
+entrypoint. In case you do not need options from flags, use
+`MakeRegistry()` in `./filters/builtin/builtin.go` to register your
+filter.
+
+
+```
+diff --git a/cmd/skipper/main.go b/cmd/skipper/main.go
```
Comment (aryszka) [inline docs/development.md]: i personally really like the code diff in the docs, never seen something like that before :+1:

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+Documentation for users should be done in `docs/`.
+
+```
+diff --git a/docs/filters.md b/docs/filters.md
+index d3bb872..a877062 100644
+--- a/docs/filters.md
++++ b/docs/filters.md
+@@ -382,6 +382,24 @@ basicAuth("/path/to/htpasswd")
+ basicAuth("/path/to/htpasswd", "My Website")
+ ```
```
Comment (aryszka) [inline docs/development.md]: not sure how the code blocks will look in the final output. Have you checked what will be generated? We have example Markdown code blocks in the diff which is already a Markdown code block. Do have any means for escaping?

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

File: docs/development.md (hunk 2 above)
Comment (szuecs) [reply docs/development.md]: it does not break the markdown

**Comment by szuecs** [comment]
:+1:

**Comment by linki** [comment]
👍

**Pull request merged**